core/vm/contracts_test.go

```diff
@@ -368,13 +368,13 @@ var proofOfPossessionTests = []precompiledTest{
 var hashHeaderTests = []precompiledTest{
 	{
 		input:         "",
-		expected:      "EOF",
+		expected:      "unable to decode input",
 		name:          "input_invalid_empty",
 		errorExpected: true,
 	},
 	{
 		input:         "f901f9a07285abd5b24742f184ad676e31f6054663b3529bc35ea2fcad8a3e0f642a46f7a01dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347948888f1f195afa192cfee860698584c030f4c9db1a0ecc60e00b3fe5ce9f6e1a10e5469764daf51f1fe93c22ec3f9a7583a80357217a0d35d334d87c0cc0a202e3756bf81fae08b1575f286c7ee7a3f8df4f0f3afc55da056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421b90100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008302000001832fefd8825208845c47775c80a0000000000000000000000000000000000000000000000000000000000000000088000000000000000",
-		expected:      "rlp: value size exceeds available input length",
```

Comment:
IMO this is a more informative error than "unable to decode input"

Reply:
See above comment about propagating errors from RLP